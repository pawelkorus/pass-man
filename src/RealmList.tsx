import React from 'react'
import { Container } from 'react-bootstrap'
import { RealmDefinition } from './service'
import RealmListItem from './RealmListItem'

type Props = {
    items:RealmDefinition[]

    onItemChanged?:(item:RealmDefinition) => void,
    onItemRemoved?:(item:RealmDefinition) => void
}

export default (props:Props):React.ReactElement => {
    return (
<Container>
    { props.items.map(item => 
    <RealmListItem
        key={item.id}
        item={item}
        onItemChanged={props?.onItemChanged}
        onItemRemoved={props?.onItemRemoved}
        />) }
</Container>
    )
}
