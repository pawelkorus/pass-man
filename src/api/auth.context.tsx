import React from 'react'
import { Authentication } from './types'

export type AuthContextProps = {
    state: {
        loading:boolean,
        authentication:Authentication
    }
}

export const AuthContext = React.createContext<AuthContextProps>({state: {loading: true, authentication: null }})
