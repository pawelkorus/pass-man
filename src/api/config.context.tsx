import React from 'react'
import { Config } from './types'

export type ConfigContextProps = {
    state: {
        loading:boolean,
        config:Config
    }
}

export const ConfigContext = React.createContext<ConfigContextProps>({state: {config: null, loading: true}})


