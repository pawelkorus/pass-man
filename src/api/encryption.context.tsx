import React from 'react'

export type Encryptor = (data:Uint8Array) => Uint8Array
export type Decryptor = (data:Uint8Array) => Uint8Array

export type EncryptionContextProps = {
    state: {
        loading:boolean
    },
    actions: {
        encrypt: Encryptor
        decrypt: Decryptor
        usePassphrase: (passphrase:String) => void
    }
}

export const EncryptionContext = React.createContext<EncryptionContextProps>({
    state: {
        loading: false
    },
    actions: {
        encrypt: (data) => {
            console.debug("encrypt data")
            return data
        },
        decrypt: (data) => {
            console.debug("decrypt data")
            return data
        },
        usePassphrase: (passphrase:String) => {
            console.debug("set passphrase " + passphrase)
        }
    }
})

export function useEncryption() {
    return React.useContext(EncryptionContext)
}


