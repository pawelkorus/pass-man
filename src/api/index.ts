export type Authentication = { __TYPE__: "Authentication" }

export type Config = { __TYPE__: "Config" }
