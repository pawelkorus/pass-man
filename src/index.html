<!DOCTYPE html>
<html>
<head>
    <title>pass-man</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="fragment" content="!">
    <meta charset="UTF-8">
</head>

<body>

    <noscript>
        This website requires Javascript to be enabled. Please turn on Javascript
        and reload the page.
    </noscript>

    <div id="root"></div>
    
</body>