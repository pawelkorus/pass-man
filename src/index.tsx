import './main.scss';
import React from 'react';
import ReactDOM from 'react-dom';

import App from './App';

/*jQuery(() => {
    Papa.parse(testCSV, {
        download: true,
        worker: true,
        step: function(row) {
            console.log("Row:", row.data);
        },
        complete: function() {
            console.log("All done!");
        }
    });
});*/

ReactDOM.render(<App />, document.querySelector('#root'));