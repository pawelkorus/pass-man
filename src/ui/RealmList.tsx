import React from 'react'
import { Container } from 'react-bootstrap'
import { RealmDefinition } from '../api'
import RealmListItem from './RealmListItem'

type Props = {
    items:RealmDefinition[]

    onItemChanged?:(item:RealmDefinition) => void,
    onItemRemoved?:(item:RealmDefinition) => void
}

export default (props:Props):React.ReactElement => {
    
    function notPersistedFirst(left:RealmDefinition, right:RealmDefinition) {
        if(left.persisted && !right.persisted) return 1
        else if(left.persisted == right.persisted) return 0
        else return -1
    }
 
    return (
<Container>
    { props.items.sort(notPersistedFirst).map(item => 
    <RealmListItem
        key={item.id}
        item={item}
        onItemChanged={props?.onItemChanged}
        onItemRemoved={props?.onItemRemoved}
        />) }
</Container>
    )
}
