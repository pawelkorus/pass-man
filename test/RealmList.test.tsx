import { unmountComponentAtNode } from 'react-dom'
import { render, screen } from "@testing-library/react";
import RealmList from "../src/RealmList"
import React from 'react';
import { RealmDefinition } from '../src/api'

it("should render empty RealmList", () => {

    let realms:RealmDefinition[] = []

    let result = render(<RealmList items={realms}></RealmList>)
    
    expect(result).toMatchSnapshot()
})

it("should render RealmList", () => {
    let realms:RealmDefinition[] = [
        { realm: "realm1", username: "username1", password: "password1", tags: [], id: "id1", persisted: true }
        ,{ realm: "realm2", username: "username2", password: "password2", tags: [], id: "id2", persisted: true }
        ,{ realm: "realm3", username: "username3", password: "password3", tags: [], id: "id3", persisted: false }
    ]

    let result = render(<RealmList items={realms}></RealmList>)

    expect(result).toMatchSnapshot()
})
