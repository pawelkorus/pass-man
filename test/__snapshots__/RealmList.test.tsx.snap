// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RealmList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="align-items-center border-bottom pt-2 pb-2 row"
        >
          <div
            class="text-break col-3"
          >
            realm1
          </div>
          <div
            class="text-break col"
          >
            username1
          </div>
          <div
            class="text-break col"
          >
            password1
          </div>
          <div
            class="text-break col-3"
          >
            <div />
          </div>
          <div
            class="col-sm-auto"
          >
            <div
              class="btn-group"
              role="group"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                <i
                  class="fas fa-edit"
                />
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                <i
                  class="fas fa-trash-alt"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="align-items-center border-bottom pt-2 pb-2 row"
        >
          <div
            class="text-break col-3"
          >
            realm2
          </div>
          <div
            class="text-break col"
          >
            username2
          </div>
          <div
            class="text-break col"
          >
            password2
          </div>
          <div
            class="text-break col-3"
          >
            <div />
          </div>
          <div
            class="col-sm-auto"
          >
            <div
              class="btn-group"
              role="group"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                <i
                  class="fas fa-edit"
                />
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                <i
                  class="fas fa-trash-alt"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="align-items-center border-bottom pt-2 pb-2 row"
        >
          <div
            class="text-break col-3"
          >
            realm3
          </div>
          <div
            class="text-break col"
          >
            username3
          </div>
          <div
            class="text-break col"
          >
            password3
          </div>
          <div
            class="text-break col-3"
          >
            <div />
          </div>
          <div
            class="col-sm-auto"
          >
            <div
              class="btn-group"
              role="group"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                <i
                  class="fas fa-edit"
                />
              </button>
              <button
                class="btn btn-danger"
                type="button"
              >
                <i
                  class="fas fa-trash-alt"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="align-items-center border-bottom pt-2 pb-2 row"
      >
        <div
          class="text-break col-3"
        >
          realm1
        </div>
        <div
          class="text-break col"
        >
          username1
        </div>
        <div
          class="text-break col"
        >
          password1
        </div>
        <div
          class="text-break col-3"
        >
          <div />
        </div>
        <div
          class="col-sm-auto"
        >
          <div
            class="btn-group"
            role="group"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              <i
                class="fas fa-edit"
              />
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              <i
                class="fas fa-trash-alt"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="align-items-center border-bottom pt-2 pb-2 row"
      >
        <div
          class="text-break col-3"
        >
          realm2
        </div>
        <div
          class="text-break col"
        >
          username2
        </div>
        <div
          class="text-break col"
        >
          password2
        </div>
        <div
          class="text-break col-3"
        >
          <div />
        </div>
        <div
          class="col-sm-auto"
        >
          <div
            class="btn-group"
            role="group"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              <i
                class="fas fa-edit"
              />
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              <i
                class="fas fa-trash-alt"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="align-items-center border-bottom pt-2 pb-2 row"
      >
        <div
          class="text-break col-3"
        >
          realm3
        </div>
        <div
          class="text-break col"
        >
          username3
        </div>
        <div
          class="text-break col"
        >
          password3
        </div>
        <div
          class="text-break col-3"
        >
          <div />
        </div>
        <div
          class="col-sm-auto"
        >
          <div
            class="btn-group"
            role="group"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              <i
                class="fas fa-edit"
              />
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              <i
                class="fas fa-trash-alt"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render empty RealmList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
